spacy/kb/candidate.pxd

```diff
@@ -1,12 +1,16 @@
-from .kb cimport KnowledgeBase
 from libcpp.vector cimport vector
+from .kb_in_memory cimport InMemoryLookupKB
 from ..typedefs cimport hash_t
 
-# Object used by the Entity Linker that summarizes one entity-alias candidate combination.
 cdef class Candidate:
-    cdef readonly KnowledgeBase kb
-    cdef hash_t entity_hash
-    cdef float entity_freq
-    cdef vector[float] entity_vector
-    cdef hash_t alias_hash
-    cdef float prior_prob
+    cdef readonly str _entity_id_
+    cdef readonly hash_t _entity_id
+    cpdef vector[float] _entity_vector
+    cdef float _prior_prob
+
+
+cdef class InMemoryCandidate(Candidate):
+    cdef readonly InMemoryLookupKB _kb
+    cdef hash_t _entity_hash
```

Comment:
Isn't `entity_hash` the same as `_entity_id` now?

Reply:
Maybe this should be a function instead?

Reply:
Ah, I forgot to remove this. It's not used and is indeed the same as `_entity_id`. Updated in https://github.com/explosion/spaCy/pull/12341/commits/27053912da0fbaae216e2d862371298228346063.